app/models/legacy_appeal.rb

```
@@ -715,8 +721,28 @@ def external_id
     vacols_id
   end
 
+  def assigned_to_location
+    return location_code unless LOCATION_CODES[:caseflow] == location_code
+
+    active_tasks = tasks.where(status: [Constants.TASK_STATUSES.in_progress, Constants.TASK_STATUSES.assigned])
+    return most_recently_assigned_to_label(active_tasks) if active_tasks.any?
+
+    on_hold_tasks = tasks.where(status: Constants.TASK_STATUSES.on_hold)
+    return most_recently_assigned_to_label(on_hold_tasks) if on_hold_tasks.any?
+
+    # shouldn't happen because if all tasks are closed the task returns to the assigning attorney
```

if this shouldn't happen, we should raise an error, not fail gracefully. It means there's bad data and we'd want to know about it.

---

This function is only used to display a label in the queue or case details page, so this situation occurring isn't terrible. I'd be in favor of sending a message to Sentry, but still returning some value here to be super safe.

---

Fair. I feel like sending "Error loading location" or something like that is better than a guess that might be wrong or misleading.

---

I think that makes sense too. I'll defer to @joeyyang about how he'd like to handle it. I think any of those three options makes sense to me (keep it as is, keep the return value as is but send the error to sentry, or use "error loading location" as the return value and send it to sentry).

---

Let's keep the return value as is and send the error to Sentry, if that works!